xbmc/pictures/Picture.cpp

```diff
@@ -277,6 +279,73 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
   return false;
 }
 
+std::unique_ptr<CTexture> CPicture::CreateTiledThumb(const std::vector<std::string>& files)
+{
+  if (!files.size())
+    return {};
+
+  unsigned int num_across = (unsigned int)ceil(sqrt((float)files.size()));
+  unsigned int num_down = (files.size() + num_across - 1) / num_across;
+
+  unsigned int imageRes = CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_imageRes;
+
+  unsigned int tile_width = imageRes / num_across;
+  unsigned int tile_height = imageRes / num_down;
+  unsigned int tile_gap = 1;
+  bool success = false;
+
+  // create a buffer for the resulting thumb
+  uint32_t* buffer = static_cast<uint32_t*>(calloc(imageRes * imageRes, 4));
```

Comment:
Why is calloc used here?

Reply:
All this is a flat copy of the original function, it just returns the texture instead of saving it directly to the file system.

Reply:
"Because it was here before" is a perfectly fine reason. Usually though we wanna use C++ allocation instead of C allocation, especially because with C++ allocation we can hide the pointer entirely with smart pointers.

Reply:
all updated.